test/framework/src/main/java/org/elasticsearch/test/ESTestCase.java

```diff
@@ -2430,4 +2431,31 @@ public static <T extends Throwable> T expectThrows(Class<T> expectedType, Reques
             () -> builder.get().decRef() // dec ref if we unexpectedly fail to not leak transport response
         );
     }
+
+    /**
+     * Run {@code numberOfTasks} parallel tasks that were created by the given {@code taskFactory}. On of the tasks will be run on the
+     * calling thread, the rest will be run on a new thread.
+     * @param numberOfTasks number of tasks to run in parallel
+     * @param taskFactory task factory
+     */
+    public static void runInParallel(int numberOfTasks, IntFunction<Runnable> taskFactory) throws InterruptedException, ExecutionException {
+        final ArrayList<Future<?>> futures = new ArrayList<>(numberOfTasks);
+        final Thread[] threads = new Thread[numberOfTasks - 1];
+        for (int i = 0; i < numberOfTasks; i++) {
+            var future = new FutureTask<Void>(taskFactory.apply(i), null);
+            futures.add(future);
+            if (i == numberOfTasks - 1) {
+                future.run();
+            } else {
+                threads[i] = new Thread(future);
+                threads[i].start();
+            }
+        }
+        for (Thread thread : threads) {
+            thread.join();
+        }
+        for (Future<?> future : futures) {
+            future.get();
```

Comment:
I'd rather we used something like `RefCountingListener` in case there are multiple exceptions to report. Or else just let any exceptions go uncaught and fail the test. Probably changes semantics a bit, not sure if it matters tho.

Reply:
Right that might actually be nice to see :) I kept it simple so we don't change what code is actually executed in this PR by adding some listener (might not matter as you say though but meh it's late and lets keep it simple :D) and just did a try-catch loop here now. Definitely an improvement over what we have today IMO